R/ospPlotConfiguration.R

```diff
@@ -0,0 +1,403 @@
+    yAxisColor = NULL,
+    yAxisSize = NULL,
+    yAxisLinetype = NULL,
+    xGridColor = NULL,
+    xGridSize = NULL,
+    xGridLinetype = NULL,
+    yGridColor = NULL,
+    yGridSize = NULL,
+    yGridLinetype = NULL,
+
+    # lines
+    linesColor = NULL,
+    linesFill = NULL,
+    linesShape = NULL,
+    linesSize = NULL,
+    linesLinetype = NULL,
+    linesAlpha = NULL,
+
+    # points
+    pointsColor = NULL,
+    pointsFill = NULL,
+    pointsShape = NULL,
+    pointsSize = NULL,
+    pointsLinetype = NULL,
+    pointsAlpha = NULL,
+
+    # ribbons
+    ribbonsColor = NULL,
+    ribbonsFill = NULL,
+    ribbonsShape = NULL,
+    ribbonsSize = NULL,
+    ribbonsLinetype = NULL,
+    ribbonsAlpha = NULL,
+
+    # errorbars
+    errorbarsColor = NULL,
+    errorbarsFill = NULL,
+    errorbarsShape = NULL,
+    errorbarsSize = NULL,
+    errorbarsLinetype = NULL,
+    errorbarsAlpha = NULL,
+
+    # export
+    plotSaveFileFormat = NULL,
+    plotSaveFileWidth = NULL,
+    plotSaveFileHeight = NULL,
+    plotSaveFileDimensionUnits = NULL,
+
+    #' @description Create a new `ospInternalPlotConfiguration` object
+    #'
+    #' @param title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+    #'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+    #'   label, plot legend, watermark, respectively.
+    #' @param legendPosition A character string defining the legend position
+    #'   (default: `"outsideRight"`). Available options
+    #'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+    #'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+    #'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+    #'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+    #'    `"outsideRight"`,`"outsideTopRight"`.
+    #' @param legendTitleSize,legendTitleColor,legendTitleFontFamily,legendTitleFontFace,legendTitleAngle,legendCaptionSize Aesthetic properties for the legend title.
+    #' @param legendCaptionColor,legendCaptionFontFamily,legendCaptionFontFace,legendCaptionAngle Aesthetic properties for the legend caption.
+    #' @param xAxisTickLabels,xAxisLabelSize,xAxisLabelColor,xAxisLabelFontFamily,xAxisLabelFontFace,xAxisLabelAngle Aesthetic properties for the x-axis label.
+    #' @param yAxisTickLabels,yAxisLabelSize,yAxisLabelColor,yAxisLabelFontFamily,yAxisLabelFontFace,yAxisLabelAngle Aesthetic properties for the y-axis label.
+    #' @param xAxisLimits,yAxisLimits A numeric vector of axis limits for the x-and
+    #'   y-axis, respectively.
+    #' @param xAxisTicks,yAxisTicks A numeric vector or a function defining where to
+    #'   position x-and y-axis ticks, respectively.
+    #' @param xAxisScale,yAxisScale A character string defining axis scale (default:
+    #'   `"lin"`). Available options are: `"lin"`, `"log"`, `"ln"`, `"discrete"`,
+    #'   `"reverse"`, `"sqrt"`, `"time"`, `"date"`.
```

Comment:
add hint that all options are accessible via `tlf::Scaling` e.g. `tlf::Scaling$lin`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html
https://www.open-systems-pharmacology.org/TLF-Library/reference/Scaling.html